### PR TITLE
[WEB-3746] Added Datadog logging

### PR DESCRIPTION
**Pull request by simonmarton**
Had quite some trouble with running Karma tests properly :(

### CODE REVIEW

File: source/javascripts/services/logger-service-instance.js.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { datadogLogs } from "@datadog/browser-logs";
+import DataDogLoggerService from "./logger-service";
+
+var loggerService = new DataDogLoggerService(datadogLogs, "<%= data[:constants][:datadog][:apiKey] %>");
+
+angular.module("BitriseWorkflowEditor").factory("loggerService", function() {
```
Comment (tadam313) [inline source/javascripts/services/logger-service-instance.js.erb]: can we call itt `logger`?

File: source/javascripts/_BitriseWorkflowEditor.js.erb [hunk 2]
```diff
@@ -68,4 +59,11 @@
 			$httpProvider.defaults.xsrfHeaderName = "X-CSRF-TOKEN";
 		}
 	]);
+
+	BitriseWorkflowEditor.factory("$exceptionHandler", function($log, loggerService) {
+		return function myExceptionHandler(error) {
+			loggerService.error(error.message, error);
+			$log.error(error);
```
Comment (tadam313) [inline source/javascripts/_BitriseWorkflowEditor.js.erb]: what does `$log` do?

File: source/javascripts/services/logger-service-instance.js.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { datadogLogs } from "@datadog/browser-logs";
+import DataDogLoggerService from "./logger-service";
+
+var loggerService = new DataDogLoggerService(datadogLogs, "<%= data[:constants][:datadog][:apiKey] %>");
```
Comment (tadam313) [inline source/javascripts/services/logger-service-instance.js.erb]: what if we say `global.logger = new DataDogLoggerService(...) ... factory(function { return global.logger; })`. So we have logger both on global and angular scope. That's nasty but at least we can mock it via karma by `global.logger = new Mock()` WDYT? Or we can also not test the logger itself, the logic is pure simply in it anyway.

File: source/javascripts/controllers/_WorkflowsController.js.erb [hunk 4]
```diff
@@ -204,6 +205,7 @@ import { safeDigest } from "../services/react-compat";
 										}
 									);
 								} catch (error) {
+									loggerService.error(error.message, error);
```
Comment (tadam313) [inline source/javascripts/controllers/_WorkflowsController.js.erb]: what does it log here? should we use `reject` part of the promise? Also couple lines above there is a `console.error` entry should we replace that as well? :)

File: source/javascripts/_BitriseWorkflowEditor.js.erb (hunk 2 above)
Comment (simonmarton) [reply source/javascripts/_BitriseWorkflowEditor.js.erb]: This is the default functionality for the exception handler, it just logs to the browser console by default.

File: source/javascripts/services/logger-service-instance.js.erb (hunk 3 above)
Comment (simonmarton) [reply source/javascripts/services/logger-service-instance.js.erb]: That's a good idea, I don't think it's that nasty, having it tested is a good trade-off. I'll try it.

File: source/javascripts/_BitriseWorkflowEditor.js.erb (hunk 2 above)
Comment (tadam313) [reply source/javascripts/_BitriseWorkflowEditor.js.erb]: do we want it in production?

Comment (simonmarton) [reply source/javascripts/_BitriseWorkflowEditor.js.erb]: I'd say so, at least until we have a stable release and monitoring solution that has proven we actually see errors on our end.

Comment (tadam313) [reply source/javascripts/_BitriseWorkflowEditor.js.erb]: ok good point

File: source/javascripts/services/logger.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	error(message: string, ctx?: Context): void {
+		this.logger.error(message, ctx);
+	}
+}
+
+// testing purposes
+(<any>window).datadogLogs = datadogLogs;
+
+export default (token: string|undefined): Logger => new DataDogLoggerService(
+    token || (<any>window).DATADOG_API_KEY,
```
Comment (tadam313) [inline source/javascripts/services/logger.ts]: @simon can you please declare it on window

**Pull request merged**